### PR TITLE
[receiver/statsd] Refactor of hardcoded tests

### PR DESCRIPTION
**Pull request by kilokang**
**Description:** 
This is the first step to adding UDS support to the StatsD receiver.

As I started to develop it, I saw that all tests were hardcoded to UDP and using networking and there was no possibility to add a socket communication.

PR started to get huge so I split it into two, one to refactor tests and another one that adds UDS support:
 * Removed all unused references.
 * Made a `Transport` helper that allows centralizing all supported protocols and constants in its package.
 * Removed all hardcoded UDP protocols and generalized testing so new protocols are easy to add.

If you need a rationale about why the changes are like this, this is the next PR I am going to submit after this one is merged: https://github.com/kang-makes/opentelemetry-collector-contrib/pull/2

That is the PR that is going to add UDS support properly.

**Link to tracking Issue:**
 - #21385

### CODE REVIEW

File: .chloggen/receiver_statsdreceiver_add-uds-support.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Use this changelog template to create an entry for release notes.
+# If your change doesn't affect end users, such as a test fix or a tooling change,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: "bug_fix"
```
Comment (dmitryax) [inline .chloggen/receiver_statsdreceiver_add-uds-support.yaml:6]: Is there a bug that fixed by this PR? If it's just refactoring, no changelog entry is needed

**Review by dmitryax — COMMENTED** [review]
@kang-makes please rebase

File: .chloggen/receiver_statsdreceiver_add-uds-support.yaml (hunk 1 above)
Comment (kang-makes) [reply .chloggen/receiver_statsdreceiver_add-uds-support.yaml]: I remove it on the commit [#7094a68](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24832/commits/7094a680995dac9e32e58c9d394fb6416fdc2a33)

**Comment by kang-makes** [comment]
Rebased :)

**Comment by kang-makes** [comment]
Rebased.

**Comment by kang-makes** [comment]
Rebased again. Could you take a look @dmitryax? I am waiting for this PR to be merged so I can create a new pull request with the feature I really want to add: The issue 21385 from the code waiting [here](https://github.com/kang-makes/opentelemetry-collector-contrib/pull/2/files) :D

**Comment by kang-makes** [comment]
@dmitryax I have just rebased it and included in this PR the support for TCP that was added by the community.

Let me know if I can do something more because I cannot reopen the PR :(

**Pull request closed without merging**